### PR TITLE
[RFR] Play the 15-puzzle in a webbrowser

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Add guzzle, twig and webprofiler as dependencies
- [x] Use the last version of the 15-puzzle 
- [x] Build a game module to communicate with the go webserver
- [x] Create a template to render the puzzle

### CODE REVIEW

**Comment by fzaninotto** [comment]
Can you add a screenshot as soon as you have something visible?

File: app/config/packages/config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+csa_guzzle:
+  clients:
+    puzzle_api:
+      config:
+        base_uri: http://15-puzzle-go:2000
```
Comment (jdemangeon) [inline app/config/packages/config.yaml]: Please use parameters to set this url

File: app/src/Api/GameSerializer.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Normalizer\ArrayDenormalizer;
+use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
+
+use App\Game\Game;
+
+class GameSerializer {
+  private $_serializer;
```
Comment (jdemangeon) [inline app/src/Api/GameSerializer.php]: s/$_serializer/$serializer/

File: app/src/Api/GameSerializer.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Normalizer\ArrayDenormalizer;
+use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
+
+use App\Game\Game;
+
+class GameSerializer {
```
Comment (jdemangeon) [inline app/src/Api/GameSerializer.php]: You can implement the SerializeInterface to "fix" class implementation
http://api.symfony.com/3.1/Symfony/Component/Serializer/SerializerInterface.html

Comment (jdemangeon) [inline app/src/Api/GameSerializer.php]: You can implement the SerializerInterface to "fix" class implementation
http://api.symfony.com/3.1/Symfony/Component/Serializer/SerializerInterface.html

File: app/src/Controller/GameController.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  private $api;
+  private $twig;
+
+  public function __construct(\Twig_Environment $twig, GameApi $gameApi) {
+    $this->api = $gameApi;
+    $this->twig = $twig;
+  }
+ 
+  public function start() {
+    $game = $this->api->new(4);
```
Comment (jdemangeon) [inline app/src/Controller/GameController.php]: Please use a constant for 4 (to know what is it)

Comment (jdemangeon) [reply app/src/Controller/GameController.php]: Can you use a linter? There's an help here https://github.com/djoos/Symfony-coding-standard

File: app/tests/Controller/MainControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Tests\App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class MainControllerTest extends WebTestCase {
+    public function testStart() {
+
```
Comment (jdemangeon) [inline app/tests/Controller/MainControllerTest.php]: Useless line

File: app/templates/main.html.twig [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+{% block header %}
+  <div class="section">
+    <h5>Welcome to the 15 puzzle game!</h5>
+  </div>
+{% endblock %}
+
+{% block content %}
+  <div class="section">
+    <a class="waves-effect waves-light btn" href="/game"><i class="material-icons left">add_circle_outline</i>Start a new game</a>
```
Comment (alexisjanvier) [inline app/templates/main.html.twig:11]: `waves-effect`, `waves-light`, you're selling dreams... :)

**Comment by Luwangel** [comment]
Main page:
capture du 2017-11-07 10-29-05

Game page:
capture du 2017-11-07 10-28-36

**Review by alexisjanvier — APPROVED** [review]

**Pull request merged**